ubuntu/common/install_doca.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+set -ex
+
+doca_metadata=$(jq -r '.doca."'"$DISTRIBUTION"'"' <<< $COMPONENT_VERSIONS)
+DOCA_VERSION=$(jq -r '.version' <<< $doca_metadata)
+DOCA_SHA256=$(jq -r '.sha256' <<< $doca_metadata)
+DOCA_URL=$(jq -r '.url' <<< $doca_metadata)
+DOCA_FILE=$(basename ${DOCA_URL})
+
+$COMMON_DIR/download_and_verify.sh $DOCA_URL $DOCA_SHA256
+
+dpkg -i $DOCA_FILE
+apt-get update
+
+apt-get -y install doca-ofed
+$COMMON_DIR/write_component_version.sh "DOCA" $DOCA_VERSION
+
+OFED_VERSION=$(ofed_info | sed -n '1,1p' | awk -F'-' 'OFS="-" {print $3,$4}' | tr -d ':')
+$COMMON_DIR/write_component_version.sh "OFED" $OFED_VERSION
+
+/etc/init.d/openibd restart
```

Comment:
Let's check openibd status to ensure it's restarted successfully.

Reply:
This is the same thing we did in the existing `install_mellanoxofed.sh` and it (I'm pretty sure) failed and exited because of the `set -ex` at the top of the file. Have you seen an issue with the existing approach?

Reply:
Yeah, I have seen it returned 0 as exit code, but openibd was not active state. It's a rare case though. 
Since IB is our most concerning thing, I think it'd be better to explicitly check its status.

Reply:
should I check the `openibd` service?

or use `lspci` or `ibstatus` like we do in the tests?

Reply:
Sure, if `/etc/init.d/openibd status` returns non-zero, when ib is inactive state.
I was thinking `systemctl is-active --quiet openibd` like what we do here
https://github.com/Azure/azhpc-images/pull/360/files#diff-9dd0c2d80493280312ebe7376360b1c8a8b664fe0269ac06f86cbbe5f6ffe067R47-R54